jbpm-wb-common/jbpm-wb-common-client/src/test/java/org/jbpm/workbench/common/client/filters/advanced/AdvancedFiltersPresenterTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.common.client.filters.advanced;
+
+import java.util.function.Consumer;
+
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import org.jbpm.workbench.common.client.filters.saved.SavedFilterSelectedEvent;
+import org.jbpm.workbench.df.client.filter.AdvancedFilterEditor;
+import org.jbpm.workbench.df.client.filter.FilterSettings;
+import org.jbpm.workbench.df.client.filter.FilterSettingsManager;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.uberfire.mocks.EventSourceMock;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+@RunWith(GwtMockitoTestRunner.class)
+public class AdvancedFiltersPresenterTest {
+
+    @Mock
+    AdvancedFilterEditor advancedFilterEditorView;
+
+    @Mock
+    FilterSettingsManager filterSettingsManager;
+
+    @Mock
+    FilterSettings filterSettingsMock;
+
+    @Mock
+    EventSourceMock<SavedFilterSelectedEvent> savedFilterEvent;
+
+    AdvancedFiltersPresenter presenter;
+
+    @Before
+    public void setup() {
+        when(filterSettingsManager.createFilterSettingsPrototype()).thenReturn(filterSettingsMock);
+        doNothing().when(savedFilterEvent).fire(any());
+        presenter = new AdvancedFiltersPresenter() {
```

Comment:
you can use _spy(AdvancedFiltersPresenter.class)_ instead of creating a inner class.